chart/templates/webserver/webserver-deployment.yaml

```diff
@@ -39,7 +39,7 @@ spec:
   strategy:
     type: RollingUpdate
     rollingUpdate:
-      maxUnavailable: 1
+      maxUnavailable: 0
```

Comment:
The original code (`maxUnavailable: 1`) already looks good to me when `replicas` >=2.

It's possible that users explicitly specify the `replicas` to 1, for some purposes. In that case, users take the responsibility and should be aware of the result.

Please correct me if I misunderstand/miss anything.

Reply:
> It's possible that users explicitly specify the replicas to 1, for some purposes. In that case, users take the responsibility and should be aware of the result.

I don't think so, now replicas defaults to 1, upgrade chart always waiting long time, this will make the webserver unavailable.
Of course, set replicas to >=2 can solve this problem, but everyone needs to do it.

Reply:
I'm not really convinced by this. I would like to have some valuable inputs from other committers as well @mik-laj @dimberman

Reply:
It makes sense for me to provide support for service updates by default, with a guarantee of its continuous availability. I don't think HA for the webserver is often deployed because the webserver is not used very heavily because the key feature are provided by the webserver and worker. The web server only supports the use of Airflow but is an optional component. The Web Server only supports the use of Airflow, but from a design perspective it is an optional, non-essential component, so in most cases, HA is not necessary.

However, to handle both cases, we can add a condition. If `replicas > 2`, then the `maxUnavailable = replicas - 1,` If `replicas == 1`, then `maxUnavailable = 0`. What do you think about it?

Reply:
@mik-laj Can we let the user decide `maxUnavailable` value?

Reply:
Can we let users decide `maxUnavailable` ?

Reply:
> Can we let users decide maxUnavailable ?
ping @mik-laj @XD-DENG

Reply:
I agree it's reasonable to let the users decide `maxUnavailable` (e.g. in `values.yaml`), **with proper default/fall-back value.**

Reply:
I agree with both.